homeassistant/components/overkiz/coordinator.py

```diff
@@ -150,14 +104,69 @@ async def _get_devices(self) -> dict[str, Device]:
         _LOGGER.debug("Fetching all devices and state via /setup/devices")
         return {d.device_url: d for d in await self.client.get_devices(refresh=True)}
 
-    def places_to_area(self, place: Place) -> dict[str, str]:
-        """Convert places with sub_places to a flat dictionary."""
+    def _places_to_area(self, place: Place) -> dict[str, str]:
+        """Convert places with sub_places to a flat dictionary [placeoid, label])."""
         areas = {}
         if isinstance(place, Place):
             areas[place.oid] = place.label
 
         if isinstance(place.sub_places, list):
             for sub_place in place.sub_places:
-                areas.update(self.places_to_area(sub_place))
+                areas.update(self._places_to_area(sub_place))
 
         return areas
+
+    @EVENT_HANDLERS.register(EventName.DEVICE_AVAILABLE)
+    async def on_device_available(self, event: Event) -> None:
```

Comment:
Since the registry is common for all coordinator instances we should not define the handlers as instance methods. Let's move the handler functions to the module level, and adjust the parameter name of the coordinator instance from `self` to `coordinator`.

Reply:
Good one! Would you prefer to have them in coordinator.py (at module level) as done in 9140767. Or would you prefer them in e.g. `event_handlers.py`? They won't be reused in any other classes / files.

Reply:
Let's keep them here.